compiler/rustc_parse/src/lib.rs

```diff
@@ -200,16 +200,23 @@ pub fn maybe_file_to_stream(
         lexer::parse_token_trees(sess, src.as_str(), source_file.start_pos, override_span);
 
     match token_trees {
-        Ok(stream) => Ok((stream, unmatched_braces)),
-        Err(err) => {
+        Ok(stream) if unmatched_braces.is_empty() => Ok((stream, unmatched_braces)),
```

Comment:
If `unmatched_braces` is always empty here, then what's the point returning it?

Reply:
`unmatched_braces` maybe not empty when returning `Ok(stream)`, `Ok(..)` means there is no `unclosed delimiter`, but there maybe unmatched_braces, for example code as below:

```rust
fn main() { )
```

`tokentrees` will return Ok(..), but `unmatched_braces` is not empty, we need to emit errors here.

Reply:
The name `unmatched_braces` and type `UnmatchedBrace` is not accurate, a proper name should be `UnmatchedDelim` since it may also contains `Bracket` and `Parenthesis` here.

I will rename it.

Reply:
Yes, I think we should refactor the code don't return `unmatched_braces`.
Done with https://github.com/rust-lang/rust/pull/108297/commits/4b5fc879c64601dd1ffe1e313fb3ad84427a25d4